### PR TITLE
Perception: make lidar height file could be specified

### PR DESCRIPTION
**Pull request by SeasoulChris**
make lidar height file could be specified

### CODE REVIEW

File: modules/perception/onboard/component/fusion_camera_detection_component.cc [hunk 1]
```diff
@@ -1141,6 +1111,46 @@ int FusionCameraDetectionComponent::MakeCameraDebugMsg(
...
+  float camera_offset = 0.0f;
+
+  apollo::perception::onboard::FusionCameraDetection
+  fusion_camera_detection_param;
+  if (!GetProtoConfig(&fusion_camera_detection_param)) {
+    AINFO << "load fusion camera detection component proto param failed";
+    return false;
+  }
+  std::string lidar_type_name_str =
+      fusion_camera_detection_param.lidar_type_name();
```
Comment (storypku) [inline modules/perception/onboard/component/fusion_camera_detection_component.cc:1128]: For readability: what about just name "lidar_type_name_str" and "lidar_type_name" to  "lidar_type" ?

File: modules/perception/onboard/proto/fusion_camera_detection_component.proto [hunk 2]
```diff
@@ -45,4 +45,5 @@ message FusionCameraDetection {
   optional bool image_based_cipv = 30 [default = false];
   optional int32 debug_level = 31 [default = 0];
   optional bool enable_cipv = 32 [default = false];
+  optional string lidar_type_name = 33 [default = "velodyne128"];
 }
```
Comment (lfcarol) [inline modules/perception/onboard/proto/fusion_camera_detection_component.proto:49]: A blank line is missing.

File: modules/perception/onboard/component/fusion_camera_detection_component.cc [hunk 3]
```diff
@@ -1141,6 +1111,46 @@ int FusionCameraDetectionComponent::MakeCameraDebugMsg(
...
+  float base_h = default_camera_height;
+  float camera_offset = 0.0f;
+
+  apollo::perception::onboard::FusionCameraDetection
+  fusion_camera_detection_param;
+  if (!GetProtoConfig(&fusion_camera_detection_param)) {
+    AINFO << "load fusion camera detection component proto param failed";
+    return false;
+  }
+  std::string lidar_type_str =
```
Comment (lfcarol) [inline modules/perception/onboard/component/fusion_camera_detection_component.cc:1127]: Please remove the suffix '_str', only 'lidar_type'.

File: modules/perception/onboard/component/fusion_camera_detection_component.cc [hunk 4]
```diff
@@ -1141,6 +1111,46 @@ int FusionCameraDetectionComponent::MakeCameraDebugMsg(
   return cyber::SUCC;
 }
 
+bool FusionCameraDetectionComponent::SetCameraHeight(
+     const std::string &sensor_name,
+     const std::string &params_dir, float default_camera_height,
+     float *camera_height) {
+  float base_h = default_camera_height;
```
Comment (lfcarol) [inline modules/perception/onboard/component/fusion_camera_detection_component.cc:1118]: You can rename it 'base_height', which is easy to realize the variable meaning.

File: modules/perception/onboard/component/fusion_camera_detection_component.cc [hunk 5]
```diff
@@ -1141,6 +1111,46 @@ int FusionCameraDetectionComponent::MakeCameraDebugMsg(
...
+  fusion_camera_detection_param;
+  if (!GetProtoConfig(&fusion_camera_detection_param)) {
+    AINFO << "load fusion camera detection component proto param failed";
+    return false;
+  }
+  std::string lidar_type =
+      fusion_camera_detection_param.lidar_type();
+  try {
+    YAML::Node lidar_height =
+        YAML::LoadFile(params_dir + "/" + lidar_type_str + "_height.yaml");
```
Comment (lfcarol) [inline modules/perception/onboard/component/fusion_camera_detection_component.cc:1131]: 'lidar_type_str' ?

**Review by jeroldchen — APPROVED** [review]

**Pull request merged**